app/services/bookings/gitis/fake_crm.rb

```diff
@@ -59,10 +63,14 @@ def update_entity(entity_id, data)
       return super unless stubbed?
       return entity_id unless entity_id.start_with?('contacts(')
 
-      data.keys.each do |key|
+      data.each do |key, value|
         unless ALLOWED.include?(key)
           raise "Bad Response - attribute '#{key}' is not recognised"
         end
+
+        if key =~ /@odata.bind\z/ && (value.blank? || value !~ Bookings::Gitis::Entity::BIND_FORMAT)
```

Comment:
As above. Also the first match could probably be made a bit easier to read using `end_with?`